CMakeLists.txt

```diff
@@ -2185,6 +2185,34 @@ list(APPEND llext_edk_cflags ${llext_filt_flags})
 list(APPEND llext_edk_cflags ${LLEXT_APPEND_FLAGS})
 list(APPEND llext_edk_cflags ${LLEXT_EDK_APPEND_FLAGS})
 
+# Define variables for the cmake sub-process
+foreach(exported_var
+        APPLICATION_SOURCE_DIR
+        CONFIG_LLEXT_EDK_NAME
+        CONFIG_LLEXT_EDK_USERSPACE_ONLY
+        llext_edk_cflags
+        llext_edk_file
+        PROJECT_BINARY_DIR
+        WEST_TOPDIR
+        ZEPHYR_BASE
+        CONFIG_LLEXT_EDK_TARGET_INFO # uses:
+            ARCH
+            BOARD
+            BOARD_REVISION
+            BOARD_QUALIFIERS
+            NORMALIZED_BOARD_QUALIFIERS
+            NORMALIZED_BOARD_TARGET
+            SOC_NAME
+            SOC_SERIES
+            SOC_FAMILY
+        CONFIG_LLEXT_EDK_TOOLCHAIN_INFO # uses:
+            CROSS_COMPILE_TARGET
+            COMPILER
+            LINKER
+        )
+  list(APPEND llext_edk_var_args -D${exported_var}="${${exported_var}}")
+endforeach()
```

Comment:
this is growing out of hands and easily becomes unmaintainable, not to mention putting undesired restrictions on Zephyr build system internals.

For the first part, starting to have a long list of settings being maintained at the toplevel CMakeLsts.txt is not ok.
What started as a small and maintainable list which is called at a time where things are known to be defined, such as: `ZEPHYR_BASE`, `WEST_TOPDIR` was acceptable, but a constant growing list for llext is not.

Secondly, exporting Zephyr internals as-is start to mean that Zephyr cannot control this.
Are we starting to promise that we will never rename those variables because they are being exported ?

Although there is no intention of renaming SOC_NAME, SOC_SERIES, BOARD_REVISION, etc. then if this change is accepted, then it sets a precedence that any variable might enter this list.

I know there is a wish for a more controlled and well-defined way of exporting Zephyr internals where names are promised to be stable, especially for CI and IDE integration.
And for that reason work is ongoing to try and establish a common output file for build info.

Perhaps this can be useful for llext to avoid above bloat in CMakeLists.txt.

Reply:
sure, it's now opened: https://github.com/zephyrproject-rtos/zephyr/pull/79118